### PR TITLE
Update index.rst for AgentQnA doc refactor

### PR DESCRIPTION
**Pull request by yinghu5**
Update index.rst for AgentQnA doc refactor

### CODE REVIEW

File: examples/AgentQnA/deploy/index.rst [hunk 1]
```diff
@@ -7,8 +7,12 @@ Here are some deployment options, depending on your hardware and environment:
...
 
 .. toctree::
    :maxdepth: 1
+   :glob:
 
-   Xeon Scalable Processor <xeon>
\ No newline at end of file
+    Xeon Scalable Processor <GenAIExamples/AgentQnA/docker_compose/intel/cpu/xeon/README>
+
+    Gaudi  <GenAIExamples/AgentQnA/docker_compose/intel/cpu/xeon/README>
```
Comment (mkbhanda) [inline examples/AgentQnA/deploy/index.rst:18]: is this a typo ... generally gaudi stuff lands in a different directory

**Review by mkbhanda — APPROVED** [review]
LGTM

File: examples/AgentQnA/deploy/index.rst (hunk 1 above)
Comment (yinghu5) [reply examples/AgentQnA/deploy/index.rst]: thank you very much for catch. fix it

File: examples/AgentQnA/AgentQnA_Guide.rst [hunk 2]
```diff
@@ -43,5 +43,13 @@ The worker agent uses the retrieval tool to generate answers to the queries post
 
 Deployment
 **********
+Here are some deployment options, depending on your hardware and environment:
+
+Single Node
```
Comment (yinghu5) [inline examples/AgentQnA/AgentQnA_Guide.rst:48]:
```suggestion
Single Node
+++++++++++++++
```

**Review by NeoZhangJianyu — APPROVED** [review]

**Pull request merged**